From 05b3ea3c10746aa454bdc05870c5aa79332332b7 Mon Sep 17 00:00:00 2001
From: jiffin tony thottan <jthottan@redhat.com>
Date: Wed, 4 Nov 2015 15:40:14 +0530
Subject: [PATCH] FSAL_GLUSTER : Adding missed check while backporting

The check in this patch was part of https://review.gerrithub.io/#/c/236924/
and it was missed while backporting.

Signed-off-by: Jiffin Tony Thottan <jthottan@redhat.com>
---
 src/FSAL/FSAL_GLUSTER/posix_acls.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/FSAL/FSAL_GLUSTER/posix_acls.c b/src/FSAL/FSAL_GLUSTER/posix_acls.c
index eb03bc1..d2cc433 100644
--- a/src/FSAL/FSAL_GLUSTER/posix_acls.c
+++ b/src/FSAL/FSAL_GLUSTER/posix_acls.c
@@ -528,6 +528,7 @@ fsal_acl_2_posix_acl(fsal_acl_t *p_fsalacl, acl_type_t type)
 
 	for (f_ace = p_fsalacl->aces;
 		f_ace < p_fsalacl->aces + p_fsalacl->naces; f_ace++) {
+		if (IS_FSAL_ACE_SPECIAL_EVERYONE(*f_ace)) {
 			if ((type == ACL_TYPE_ACCESS &&
 			!is_ace_valid_for_effective_acl_entry(f_ace))
 			|| (type == ACL_TYPE_DEFAULT &&
@@ -549,6 +550,7 @@ fsal_acl_2_posix_acl(fsal_acl_t *p_fsalacl, acl_type_t type)
 				if (IS_FSAL_ACE_EXECUTE(*f_ace) && !deny_e_x)
 					acl_add_perm(e_a_permset, ACL_EXECUTE);
 			}
+		}
 	}
 
 	/*
-- 
2.1.0

